packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -2615,6 +2598,42 @@ export class ProjectService extends BaseService {
             userAttributes,
             this.lightdashConfig.query.timezone || 'UTC',
         );
+        // Add a cache_autocomplete prefix to the query hash to avoid collisions with the results cache
+        const queryHashKey = metricQuery.timezone
+            ? `${projectUuid}.cache_autocomplete.${query}.${metricQuery.timezone}`
+            : `${projectUuid}.cache_autocomplete.${query}`;
+        const queryHash = crypto
+            .createHash('sha256')
+            .update(queryHashKey)
+            .digest('hex');
+
+        const isCacheEnabled =
+            this.lightdashConfig.resultsCache.autocompleteEnabled &&
+            this.s3CacheClient.isEnabled();
+
+        if (!forceRefresh && isCacheEnabled) {
+            const isCached = await this.s3CacheClient.getResultsMetadata(
+                queryHash,
+            );
+
+            if (isCached !== undefined) {
+                const cacheEntry = await this.s3CacheClient.getResults(
+                    queryHash,
+                );
+                const stringResults =
+                    await cacheEntry.Body?.transformToString();
+                if (stringResults) {
+                    try {
+                        return JSON.parse(stringResults);
```

Comment:
Response doesn't have`cached: true,` anymore. is that expected?

Reply:
When we upload the results to s3, we include all the metadata , like cached, or refreshedAt. When  fetching from s3, we simply return the same object we previously store. 

To avoid conflicts with normal cached results,  thes are stored in a different cacheKey